File: no-sdk/dotnet/samples/BedrockConverseSample.cs
using System.Text;
using System.Text.Json;

public class BedrockConverseSample
{
    const string SERVICE = "bedrock";
    public static void Run(string region, string modelId, string prompt)
    {
        var requestUrl = $"https://bedrock-runtime.{region}.amazonaws.com/model/{modelId}/converse";
        var payload = JsonSerializer.Serialize(new BedrockConverseRequest(prompt), options: new JsonSerializerOptions(JsonSerializerDefaults.Web));
        var responseBody = HttpHelpers.Post(SERVICE, region, requestUrl, payload);
        Console.WriteLine(responseBody);
    }
}

class BedrockConverseRequest
{
    public BedrockConverseRequest(string prompt)
    {
        Messages = new List<BedrockConverseRequestMessage>{
            new BedrockConverseRequestMessage{
                Content = new List<BedrockConverseRequestMessageContent>{
                    new BedrockConverseRequestMessageContent{
                        Text = prompt
                    }
                },
                Role = "user"
            }
        };
    }
    public List<BedrockConverseRequestMessage> Messages { get; set; }
}

class BedrockConverseRequestMessage
{
    public string Role { get; set; }
    public List<BedrockConverseRequestMessageContent> Content { get; set; }
}

class BedrockConverseRequestMessageContent
{
    public string Text { get; set; }
}

File: no-sdk/java/samples/BedrockConverseSample.java
package sigv4.signing.samples;

import sigv4.signing.HttpHelpers;

public class BedrockConverseSample {
    private static final String SERVICE = "bedrock";

    public static void run(String region, String modelId, String prompt) {
        try {
            String requestUrl = String.format("https://bedrock-runtime.%s.amazonaws.com/model/%s/converse", region, modelId);
            String payload = String.format("{\"messages\":[{\"role\":\"user\",\"content\":[{\"text\":\"%s\"}]}]}", prompt);
            String responseBody = HttpHelpers.post(SERVICE, region, requestUrl, payload);
            System.out.println(responseBody);
        } catch (Exception e) {
            System.err.println("Error in BedrockConverseSample: " + e.getMessage());
            e.printStackTrace();
        }
    }
}
